types/protractor-beautiful-reporter/index.d.ts

```diff
@@ -0,0 +1,92 @@
+// Type definitions for Angularized HTML Reporter with Screenshots for Protractor 1.3
+// Project: https://github.com/Evilweed/protractor-beautiful-reporter
+// Definitions by: Adam Kwiatek <https://github.com/akwiatek>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import jasmine = require('jasmine');
+import { Capabilities } from 'selenium-webdriver';
+
+export = HtmlReporter;
+
+declare class HtmlReporter {
+    constructor(options: HtmlReporter.HtmlReporterConstructorOptions);
+    getJasmine2Reporter(): jasmine.CustomReporter;
+    jasmine2MetaDataBuilder(
+        spec: null,
+        descriptions: string[],
+        result: jasmine.SpecResult,
+        capabilities: Capabilities,
+    ): HtmlReporter.StubMetaData;
+}
+
+declare namespace HtmlReporter {
+    interface HtmlReporterConstructorOptions {
+        baseDirectory: string;
+        pathBuilder?: (
+            spec: null,
+            descriptions: string[],
+            result: jasmine.SpecResult,
+            capabilities: Capabilities,
+        ) => string;
```

Comment:
Based on https://github.com/Evilweed/protractor-beautiful-reporter/blob/c2c22a71eabff89ea6d8b08c1b77dbf84edd86cd/index.js#L4762

Please note that it says that spec is an object but actually null specs are passed to both builder functions
https://github.com/Evilweed/protractor-beautiful-reporter/blob/c2c22a71eabff89ea6d8b08c1b77dbf84edd86cd/app/reporter.js#L271